int/config/network.go

```diff
@@ -166,16 +135,33 @@ func (n *NetworkManager) SwitchNetwork(selectedNetworkStr string) error {
 		return fmt.Errorf("unknown network option: %s", selectedNetworkStr)
 	}
 
-	version, err := n.version(cfg.URLs[0])
+	url := cfg.URLs[0]
+	client := node.NewClient(url)
+
+	status, err := node.Status(client)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve node status: %w", err)
+	}
+
+	version, err := versionFromStatus(status)
 	if err != nil {
 		return fmt.Errorf("getting network version: %w", err)
 	}
 
+	// compare chain id from node status with chain id from config
+	nodeChainID := uint64(*status.ChainID)
+	if nodeChainID != cfg.ChainID {
+		logger.Errorf("chain id mismatch: %d != %d", nodeChainID, cfg.ChainID)
```

Comment:
should we exit here?

```suggestion
		logger.Fatalf("chain id mismatch: %d != %d", nodeChainID, cfg.ChainID)
```

Reply:
I guess we shouldn't exit. It's a node issue. 
But we could display an error (or a warning) to tell user that there is something weird with the connected node and so, that we recommand switching node.

Maybe that should be done in another issue/PR as I don't think it is a priority